bmp180/bmp180.go

```diff
@@ -0,0 +1,180 @@
+}
+
+// Device wraps an I2C connection to a BMP180 device.
+type Device struct {
+	bus                     machine.I2C
+	mode                    OversamplingMode
+	calibrationCoefficients calibrationCoefficients
+}
+
+// New creates a new BMP180 connection. The I2C bus must already be
+// configured.
+//
+// This function only creates the Device object, it does not touch the device.
+func New(bus machine.I2C) Device {
+	return Device{
+		bus:  bus,
+		mode: ULTRAHIGHRESOLUTION,
+	}
+}
+
+// Connected returns whether a BMP180 has been found.
+// It does a "who am I" request and checks the response.
+func (d *Device) Connected() bool {
+	data := []byte{0}
+	d.bus.ReadRegister(Address, WHO_AM_I, data)
+	return data[0] == CHIP_ID
+}
+
+// Configure sets up the device for communication and
+// read the calibration coefficientes.
+func (d *Device) Configure() {
+	data := make([]byte, 22)
+	err := d.bus.ReadRegister(Address, AC1_MSB, data)
+	if err != nil {
+		return
+	}
+	d.calibrationCoefficients.ac1 = readInt(data[0], data[1])
+	d.calibrationCoefficients.ac2 = readInt(data[2], data[3])
+	d.calibrationCoefficients.ac3 = readInt(data[4], data[5])
+	d.calibrationCoefficients.ac4 = readUint(data[6], data[7])
+	d.calibrationCoefficients.ac5 = readUint(data[8], data[9])
+	d.calibrationCoefficients.ac6 = readUint(data[10], data[11])
+	d.calibrationCoefficients.b1 = readInt(data[12], data[13])
+	d.calibrationCoefficients.b2 = readInt(data[14], data[15])
+	d.calibrationCoefficients.mb = readInt(data[16], data[17])
+	d.calibrationCoefficients.mc = readInt(data[18], data[19])
+	d.calibrationCoefficients.md = readInt(data[20], data[21])
+}
+
+// Temperature returns the temperature in celsius milli degrees (ºC/1000)
+func (d *Device) Temperature() (temperature int32, err error) {
+	rawTemp, err := d.rawTemp()
+	if err != nil {
+		return
+	}
+	b5 := d.calculateB5(rawTemp)
+	t := (b5 + 8) >> 4
+	return 100*t, nil
+}
+
+// Pressure returns the pressure in milli pascals (mPa)
+func (d *Device) Pressure() (pressure int32, err error) {
+	rawTemp, err := d.rawTemp()
+	if err != nil {
+		return
+	}
+	rawPressure, err := d.rawPressure(d.mode)
+	if err != nil {
+		return
+	}
+	b5 := d.calculateB5(rawTemp)
+	b6 := b5 - 4000
+	x1 := (int32(d.calibrationCoefficients.b2) * (b6 * b6 >> 12)) >> 11
+	x2 := (int32(d.calibrationCoefficients.ac2) * b6) >> 11
+	x3 := x1 + x2
+	b3 := (((int32(d.calibrationCoefficients.ac1)*4 + x3) << uint(d.mode)) + 2) >> 2
+	x1 = (int32(d.calibrationCoefficients.ac3) * b6) >> 13
+	x2 = (int32(d.calibrationCoefficients.b1) * ((b6 * b6) >> 12)) >> 16
+	x3 = ((x1 + x2) + 2) >> 2
+	b4 := (uint32(d.calibrationCoefficients.ac4) * uint32(x3+32768)) >> 15
+	b7 := uint32(rawPressure-b3) * (50000 >> uint(d.mode))
+	var p int32
+	if b7 < 0x80000000 {
+		p = int32((b7 << 1) / b4)
+	} else {
+		p = int32((b7 / b4) << 1)
+	}
+	x1 = (p >> 8) * (p >> 8)
+	x1 = (x1 * 3038) >> 16
+	x2 = (-7357 * p) >> 16
+	return int32(1000*float32(p + ((x1 + x2 + 3791) >> 4))), nil
```

Comment:
Yes, will change that. I was thinking to return Pa isntead of mPa. The common unit used is hPa, and the sensor measures in Pa, which gives us 2 (instead of 3) digit precision. I think it's enough.

```go
  return 1000*(p + ((x1 + x2 + 3791) >> 4)), nil
```

or 



```go
  return p + ((x1 + x2 + 3791) >> 4), nil
```

I'm ok with both

Reply:
I would go with mPa. This particular sensor may be measuring with Pa precision, but there are probably other sensors that measure in a higher precision. By returning mPa, those other sensors can return measurements with a higher precision using the same interface.

Of course, you may change the example to show Pa instead of mPa, that's totally up to you.